Examples/Python/src/Geant4Component.cpp

```diff
@@ -51,196 +36,89 @@ using namespace pybind11::literals;
-        }
-
-        // An ACTS TrackingGeometry is provided, so simulation for sensitive
-        // detectors is turned on - they need to get matched first
-        if (trackingGeometry) {
-          SensitiveSurfaceMapper::Config ssmCfg;
-          ssmCfg.trackingGeometry = trackingGeometry;
-
-          // Take the default args if nothing provided
-          if (not volumeMappings.empty()) {
-            ssmCfg.volumeMappings = volumeMappings;
-          }
-          if (not materialMappings.empty()) {
-            ssmCfg.materialMappings = materialMappings;
-          }
-
-          g4Cfg.sensitiveSurfaceMapper =
-              std::make_shared<const SensitiveSurfaceMapper>(
-                  ssmCfg, logger->cloneWithSuffix("SensitiveSurfaceMapper"));
-        }
-
-        return g4Cfg;
-      },
-      "level"_a, "detector"_a, "randomNumbers"_a, "inputParticles"_a,
-      py::arg("trackingGeometry") = nullptr, py::arg("magneticField") = nullptr,
-      py::arg("volumeMappings") = std::vector<std::string>{},
-      py::arg("materialMappings") = std::vector<std::string>{},
-      py::arg("killVolume") = nullptr,
-      py::arg("killAfterTime") = std::numeric_limits<double>::infinity(),
-      py::arg("recordHitsOfSecondaries") = true,
-      py::arg("keepParticlesWithoutHits") = true,
-      py::arg("physicsList") = "FTFP_BERT");
+  {
+    using Algorithm = Geant4Simulation;
+    using Config = Algorithm::Config;
+    auto alg =
+        py::class_<Algorithm, ActsExamples::IAlgorithm,
+                   std::shared_ptr<Algorithm>>(mod, "Geant4Simulation")
+            .def(py::init<const Config&, Acts::Logging::Level>(),
+                 py::arg("config"), py::arg("level"))
+            .def_property_readonly("config", &Algorithm::config)
+            .def_property_readonly("geant4Handle", &Algorithm::geant4Handle);
+
+    auto c1 = py::class_<Config, std::shared_ptr<Config>>(alg, "Config")
+                  .def(py::init<>());
+    ACTS_PYTHON_STRUCT_BEGIN(c1, Config);
+    ACTS_PYTHON_MEMBER(inputParticles);
+    ACTS_PYTHON_MEMBER(randomNumbers);
+    ACTS_PYTHON_MEMBER(detectorConstructionFactory);
+    ACTS_PYTHON_MEMBER(geant4Handle);
+    ACTS_PYTHON_MEMBER(outputSimHits);
+    ACTS_PYTHON_MEMBER(outputParticlesInitial);
+    ACTS_PYTHON_MEMBER(outputParticlesFinal);
+    ACTS_PYTHON_MEMBER(trackingGeometry);
+    ACTS_PYTHON_MEMBER(magneticField);
+    ACTS_PYTHON_MEMBER(physicsList);
+    ACTS_PYTHON_MEMBER(volumeMappings);
+    ACTS_PYTHON_MEMBER(materialMappings);
+    ACTS_PYTHON_MEMBER(killVolume);
+    ACTS_PYTHON_MEMBER(killAfterTime);
+    ACTS_PYTHON_MEMBER(recordHitsOfSecondaries);
+    ACTS_PYTHON_MEMBER(keepParticlesWithoutHits);
+    ACTS_PYTHON_STRUCT_END();
+  }
 
   {
-    using Detector = ActsExamples::Telescope::TelescopeDetector;
-    using DetectorConstruction =
-        ActsExamples::Telescope::TelescopeG4DetectorConstruction;
+    using Algorithm = Geant4MaterialRecording;
```

Comment:
Probably this algorithm should work with the declare-algorithm-macro. But is there a `geant4Handle` accessor missing?